File: app/src/main/java/com/coinninja/coinkeeper/cn/service/CNFirebaseMessagingService.java
package com.coinninja.coinkeeper.cn.service;

import com.coinninja.coinkeeper.util.CNLogger;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class CNFirebaseMessagingService extends FirebaseMessagingService {

    @Inject
    PushNotificationServiceManager pushNotificationServiceManager;

    @Inject
    CNLogger logger;

    @Override
    public void onCreate() {
        AndroidInjection.inject(this);
        super.onCreate();
    }

    @Override
    public void onNewToken(String token) {
        super.onNewToken(token);
        pushNotificationServiceManager.saveToken(token);
    }


    @Override
    public void onMessageSent(String s) {
        logger.debug("----- firebase messenger", "----- message sent");
        super.onMessageSent(s);
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        logger.debug("----- firebase messenger", "----- received message");
        super.onMessageReceived(remoteMessage);
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/cn/service/CNFirebaseMessagingService.kt
package com.coinninja.coinkeeper.cn.service

import com.coinninja.coinkeeper.util.CNLogger
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import dagger.android.AndroidInjection
import javax.inject.Inject

class CNFirebaseMessagingService : FirebaseMessagingService() {
    @Inject
    internal lateinit var pushNotificationServiceManager: PushNotificationServiceManager
    @Inject
    internal lateinit var logger: CNLogger

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        pushNotificationServiceManager.saveToken(token)
    }

    override fun onMessageSent(s: String) {
        logger.debug("----- firebase messenger", "----- message sent")
        super.onMessageSent(s)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        logger.debug("----- firebase messenger", "----- received message")
        super.onMessageReceived(remoteMessage)
    }
}